src/libcore/slice/mod.rs

```diff
@@ -2414,30 +2414,16 @@ impl<'a, T> Iterator for ExactChunks<'a, T> {
             self.v = &[];
             None
         } else {
-            let end = match start.checked_add(self.chunk_size) {
-                Some(sum) => cmp::min(self.v.len(), sum),
-                None => self.v.len(),
-            };
-
-            if end - start != self.chunk_size {
-                self.v = &[];
-                None
-            } else {
-                let nth = &self.v[start..end];
-                self.v = &self.v[end..];
-                Some(nth)
-            }
+            let (_, snd) = self.v.split_at(start);
+            self.v = snd;
+            assert!(self.v.len() == self.chunk_size);
+            self.next()
```

Comment:
Why the assertion? It doesn't look correct as written, maybe >= was intended?

I'd probably avoid the assertion, there will be a bounds check equivalent to it anyhow in next(?).

Reply:
Indeed, I was confused. Thanks!